### PR TITLE
#494 - RepoPermissions.FromSettings fully implemented

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #494 and #497 
Artifactory method `PUT /api/security/permissions/{target}` request body json contains users and permissions in the following [format](https://www.jfrog.com/confluence/display/JFROG/Security+Configuration+JSON)
```
...
"users" : {
  "bob": ["read","write","manage"],
  "alice" : ["write","annotate", "read"]
}
...
```
So, I've
- introduced `UserPermission` class and changed `RepoPermissions#addUpdate` and `RepoPermissions#permissions` signature
- implemented and tested `RepoPermissions#addUpdate`
- implemented `RepoPermissions#remove` as it's part of `RepoPermissions#addUpdate`

### CODE REVIEW

File: src/main/java/com/artipie/RepoPermissions.java [hunk 1]
```diff
@@ -61,25 +67,34 @@
     /**
      * Adds or updates repository permission.
      * @param repo Repository name
-     * @param username Username
-     * @param permission Permission name
+     * @param permissions Permissions list
      * @return Completion action
      */
-    CompletionStage<Void> addUpdate(String repo, String username, String permission);
+    CompletionStage<Void> addUpdate(String repo, List<UserPermission> permissions);
```
Comment (olegmoz) [inline src/main/java/com/artipie/RepoPermissions.java:73]: @olenagerasimova I am concerned about `List<UserPermission>`. It makes one think that users are ordered and order does matter. I do not think that this is the case and would suggest to change it to `Collection`. Same goes for `permissions` method return value

File: src/test/java/com/artipie/RepoPermissionsFromSettingsTest.java [hunk 2]
```diff
@@ -103,6 +110,88 @@ void returnsEmptyMapWhenPermissionsAreNotSet() {
         );
     }
 
+    @Test
+    void updatesUserPermissions() throws IOException {
```
Comment (olegmoz) [inline src/test/java/com/artipie/RepoPermissionsFromSettingsTest.java:114]: @olenagerasimova I think we miss the case when user has some permissions and we update the user by adding more permissions or removing some permissions

File: src/test/java/com/artipie/RepoPermissionsFromSettingsTest.java [hunk 3]
```diff
@@ -103,6 +110,88 @@ void returnsEmptyMapWhenPermissionsAreNotSet() {
...
+            )
+            .toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            this.permissionsForUser(repo, ann),
+            Matchers.contains(download)
+        );
+    }
+
+    @Test
+    void deletesPermissionSection() throws IOException {
```
Comment (olegmoz) [inline src/test/java/com/artipie/RepoPermissionsFromSettingsTest.java:165]: @olenagerasimova this test will pass if `remove` method just clears all repositories. May we check that other repos stay intact after removing a repo?

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see the comments

**Comment by olenagerasimova** [comment]
@olegmoz thanks, all fixed

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#506` is not assigned, can't get performer

**Pull request merged**